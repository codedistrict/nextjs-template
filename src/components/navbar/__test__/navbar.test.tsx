import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';
import React from 'react';

import Navbar from '../index';

// Unit test
test('renders Navbar component with links', () => {
  render(<Navbar />);

  const homeLink = screen.getByText(/home/i);
  const aboutLink = screen.getByText(/about/i);
  const externalLink = screen.getByText(/external link/i);
  const links = screen.getAllByRole('link');

  expect(homeLink).toBeInTheDocument();
  expect(aboutLink).toBeInTheDocument();
  expect(externalLink).toBeInTheDocument();
  expect(links).toHaveLength(3);
});

// Snapshot test
test('matches snapshot', () => {
  render(<Navbar />);

  expect(screen.getByTestId('navbar')).toMatchSnapshot();
});
