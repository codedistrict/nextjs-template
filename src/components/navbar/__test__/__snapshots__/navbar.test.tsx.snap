// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  class="navbar"
  data-testid="navbar"
>
  <ul>
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/about"
      >
        About
      </a>
    </li>
    <li>
      <a
        href="https://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        External Link
      </a>
    </li>
  </ul>
</nav>
`;
